src/agri_ai/core/agent.py

```
@@ -3,51 +3,106 @@
 
 import logging
-from typing import Dict, Any, List, Optional
+import time
+import asyncio
+from typing import Dict, Any, List, Optional, Protocol
 from langchain.agents import AgentType, initialize_agent
-from langchain.chat_models import ChatOpenAI
+from langchain_google_genai import ChatGoogleGenerativeAI
+from langchain_openai import ChatOpenAI
 from langchain.memory import ConversationBufferMemory
 from langchain.tools import BaseTool
```

Ruff (0.11.9)

`langchain.tools.BaseTool` imported but unused

Remove unused import: `langchain.tools.BaseTool`

(F401)